src/H5FDmpi.c

```diff
@@ -252,6 +252,43 @@ H5FD_mpi_haddr_to_MPIOff(haddr_t addr, MPI_Offset *mpi_off /*out*/)
     FUNC_LEAVE_NOAPI(ret_value)
 }
 
+/*-------------------------------------------------------------------------
+ * Function:	H5FD_mpi_get_file_sync_required
+ *
+ * Purpose:	Retrieves the mpi_file_sync_required used for the file
+ *
+ * Return:	Success:	Non-negative
+ *
+ *              Failure:	Negative
+ *
+ * Programmer:	Houjun Tang
+ *              May 19, 2022
+ *
+ *-------------------------------------------------------------------------
+ */
+herr_t
+H5FD_mpi_get_file_sync_required(H5FD_t *file, hbool_t *file_sync_required)
+{
+    const H5FD_class_t *cls;
+    uint64_t            flags = H5FD_CTL_FAIL_IF_UNKNOWN_FLAG | H5FD_CTL_ROUTE_TO_TERMINAL_VFD_FLAG;
```

Comment:
Thinking about it more, I'm hesitant to use the "fail if unknown" flag here, because it doesn't seem like something we necessarily want to force every MPI-based VFD to implement. Or at the very least, a failure due to an unknown op code should be gracefully handled here, but that's a little trickier since there isn't a good way to distinguish between that and real failures currently.

Reply:
@jhendersonHDF , just changed the previous goto_error and removed the "fail if unknown" flag.